argus/backend/service/notification_manager.py

```diff
@@ -11,19 +13,53 @@ class NotificationManagerException(Exception):
 
 
 class NotificationManagerService:
+
+    def __init__(self) -> None:
+        self.notification_services = []
+        self.attach_notification_sender(ArgusDBNotificationSaver())
+        self.attach_notification_sender(EmailNotificationServiceSender())
+
+    def attach_notification_sender(self, service):
+        self.notification_services.append(service)
+
+    def send_notification(self, receiver: UUID, sender: UUID,
+                          notification_type: ArgusNotificationTypes,
+                          source_type: ArgusNotificationSourceTypes,
+                          source_id: UUID, title: str | None = None,
+                          content: str | None = None, content_params: dict | None = None) -> Any:
+
+        for service in self.notification_services:
+            service.send_notification(receiver, sender, notification_type, source_type,
+                                      source_id, title, content, content_params)
+
+    def get_notificaton(self, receiver: UUID, notification_id: UUID) -> ArgusNotification:
+        return ArgusNotification.get(receiver=receiver, id=notification_id)
+
+    def get_unread_count(self, receiver: UUID) -> int:
+        query = ArgusNotification.filter(
+            receiver=receiver, state__eq=ArgusNotificationState.UNREAD.value).allow_filtering().all()
+        return len(query)
+
+    def read_notification(self, receiver: UUID, notification_id: UUID) -> ArgusNotification:
+        notification = ArgusNotification.get(receiver=receiver, id=notification_id)
+        notification.state = ArgusNotificationState.READ
+        return bool(notification.save().state)
+
+    def get_notifications(self, receiver: UUID, limit: int = 20, after: UUID | None = None) -> list[ArgusNotification]:
```

Comment:
Both supported from 3.10, why do you think it is better?

Reply:
if it is supported by 3.10 then it is just my preference. You can ignore this comment.

Reply:
You should use native types where possible instead of aliases, they are more useful when doing reflection on function signatures, plus we're using 3.10+ anyway.